src/renderer/screens/Assets/components/NetworkAssets/NetworkAssets.tsx

```diff
@@ -41,15 +44,34 @@ export const NetworkAssets = ({
   const { getLiveNetworkBalances } = useBalance();
 
   const [filteredAssets, setFilteredAssets] = useState<Asset[]>([]);
+  const [balancesObject, setBalancesObject] = useState<Record<string, Balance>>({});
+
+  const accountIds = useMemo(
+    () =>
+      accounts.reduce<AccountId[]>((acc, account) => {
+        if (!account.chainId || account.chainId === chain.chainId) acc.push(account.accountId);
+
+        return acc;
+      }, []),
+    [chain.chainId, accounts.map((a) => a.accountId).join('')],
+  );
 
   const balances = getLiveNetworkBalances(accountIds, chain.chainId);
 
-  const balancesObject =
-    balances?.reduce<Record<string, Balance>>((acc, balance) => {
-      acc[balance.assetId] = sumBalances(balance, acc[balance.assetId]);
+  useEffect(() => {
+    const newBalancesObject = Object.values(groupBy(balances, 'assetId')).reduce<Record<string, Balance>>(
+      (acc, balances: BalanceDS[]) => {
+        if (balances.length !== new Set(accountIds).size) return acc;
```

Comment:
`new Set(accountIds).size` can be moved out of `newBalancesObject` so it's not being calculated every iteration 😉
We can invert the condition so `return acc` will present only once